### DISCUSSION TITLE
Apple M1 - Can't Install Tensorflow

### DISCUSSION THREAD

**Discussion by twopiearr** [vladmandic/sdnext#392] [General]
Just installed onto an Apple M1 MacBookAir. Watched things, and both on install and subsequent attempts to start, I see the following:

`15:15:07-127350 ERROR    Error running pip with args: install --upgrade tensorflow==2.12.0`

The interface launches, but whenever I try to generate something I get:

`Traceback (most recent call last):
  File "/Users/$USER/automatic/venv/lib/python3.9/site-packages/gradio/routes.py", line 394, in run_predict
    output = await app.get_blocks().process_api(
  File "/Users/$USER/automatic/venv/lib/python3.9/site-packages/gradio/blocks.py", line 1073, in process_api
    inputs = self.preprocess_data(fn_index, inputs, state)
  File "/Users/$USER/automatic/venv/lib/python3.9/site-packages/gradio/blocks.py", line 962, in preprocess_data
    processed_input.append(block.preprocess(inputs[i]))
IndexError: list index out of range
Traceback (most recent call last):
  File "/Users/$USER/automatic/venv/lib/python3.9/site-packages/gradio/routes.py", line 394, in run_predict
    output = await app.get_blocks().process_api(
  File "/Users/$USER/automatic/venv/lib/python3.9/site-packages/gradio/blocks.py", line 1073, in process_api
    inputs = self.preprocess_data(fn_index, inputs, state)
  File "/Users/$USER/automatic/venv/lib/python3.9/site-packages/gradio/blocks.py", line 962, in preprocess_data
    processed_input.append(block.preprocess(inputs[i]))
IndexError: list index out of range
Available models: /Users/$USER/automatic/models/Stable-diffusion 1
Initializing ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% -:--:-- 0:00:00loc("mps_add"("(mpsFileLoc): /AppleInternal/Library/BuildRoots/9e200cfa-7d96-11ed-886f-a23c4f261b56/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShadersGraph/mpsgraph/MetalPerformanceShadersGraph/Core/Files/MPSGraphUtilities.mm":228:0)): error: input types 'tensor<2x1280xf32>' and 'tensor<1280xf16>' are not broadcast compatible
LLVM ERROR: Failed to infer result type(s).
zsh: abort      ./webui.sh`

After poking around on the PIP website, I tried just invoking `pip install tensorflow` and got the following error:

`ERROR: Could not find a version that satisfies the requirement tensorflow (from versions: none)
ERROR: No matching distribution found for tensorflow`

I'm a seasoned vet when it comes to computers but new to python. Any thoughts on how to proceed from here? The main fork has so many problems on Mac and I was hoping that since this one is being updated it might work better. There does seem to be a Mac version of Tensorflow on the PIP website, but I'm afraid I don't know how to build wheels manually.

**Comment by vladmandic** [comment]
see #347

**Reply by twopiearr** [reply]
Beautiful - applied those settings then nuked my venv and let it rebuild, and now it works.
